events/events.go

```diff
@@ -404,44 +403,150 @@ type OperationResizedData struct {
 	Region   *Region      `json:"region,omitempty"`
 }
 
+// OperationFailed represents a resize operation event.
+type OperationFailed struct {
+	BaseBody
+	Data OperationFailedData `json:"data"`
+}
+
+// OperationFailedData holds the event specific data.
+type OperationFailedData struct {
+	OperationID manifold.ID `json:"operation_id"`
+	Operation   *Operation  `json:"operation"`
+
+	ResourceID *manifold.ID `json:"resource_id,omitempty"`
+	Resource   *Resource    `json:"resource,omitempty"`
+
+	UserID *manifold.ID `json:"user_id,omitempty"`
+	User   *User        `json:"user,omitempty"`
+
+	TeamID *manifold.ID `json:"team_id,omitempty"`
+	Team   *Team        `json:"team,omitempty"`
+
+	ProjectID *manifold.ID `json:"project_id,omitempty"`
+	Project   *Project     `json:"project,omitempty"`
+
+	ProviderID *manifold.ID `json:"provider_id,omitempty"`
+	Provider   *Provider    `json:"provider,omitempty"`
+
+	ProductID *manifold.ID `json:"product_id,omitempty"`
+	Product   *Product     `json:"product,omitempty"`
+
+	PlanID *manifold.ID `json:"plan_id,omitempty"`
+	Plan   *Plan        `json:"plan,omitempty"`
+
+	RegionID *manifold.ID `json:"region_id,omitempty"`
+	Region   *Region      `json:"region,omitempty"`
+
+	Error OperationError `json:"error"`
+}
+
 // User is a simplified version for events data.
 type User struct {
 	ID    manifold.ID `json:"id"`
 	Name  string      `json:"name"`
 	Email string      `json:"email"`
 }
 
+// Validate returns whether or not the given User is valid
+func (User) Validate(v interface{}) error {
+	return nil
```

Comment:
What are all these validates for, why do they all return nil?

Reply:
As explained, the validate function is required for go-swagger to be able to parse the struct. There is nothing to be validated, I'm not sure if there is a better way to skip that requirement.